src/providers/wms/qgswmscapabilities.cpp

```diff
@@ -905,6 +905,17 @@ void QgsWmsCapabilities::parseLayer( QDomElement const & e, QgsWmsLayerProperty&
 
         parseStyle( e1, styleProperty );
 
+        for ( int i = 0; i < layerProperty.style.size(); ++i )
+        {
+          if ( layerProperty.style[i].name == styleProperty.name )
```

Comment:
@pvalsecc I think it's just good practice to get into. In some cases it can make a huge difference, so I've been looking into a static analyser check to flag this on Travis. Better to fix it now then make more work for me later ;)